### PR TITLE
[ICP-9107] Convert BigDecimal to Double.

### PR DESCRIPTION
**Pull request by MGoralczykS**
@greens @tpmanley @MMateusiakS 

Could you review this?

### CODE REVIEW

**Comment by greens** [comment]
If this fixes the issue, can you also fix it in all the other public DTHs that do the same?

File: devicetypes/fibargroup/fibaro-double-switch-2-zw5.src/fibaro-double-switch-2-zw5.groovy [hunk 1]
```diff
@@ -330,23 +330,25 @@ def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd, ep=null) {
...
 			case 0:
-				sendEvent([name: "energy", value: cmd.scaledMeterValue, unit: "kWh"])
+				sendEvent([name: "energy", value: value, unit: "kWh"])
 				break
 			case 2:
-				sendEvent([name: "power", value: cmd.scaledMeterValue, unit: "W"])
+				sendEvent([name: "power", value: value, unit: "W"])
 				break
 		}
 		multiStatusEvent("${(device.currentValue("power") ?: "0.0")} W | ${(device.currentValue("energy") ?: "0.00")} kWh")
```
Comment (tpmanley) [inline devicetypes/fibargroup/fibaro-double-switch-2-zw5.src/fibaro-double-switch-2-zw5.groovy]: Should update the default values used here too

**Comment by MGoralczykS** [comment]
@greens @tpmanley 
Could you check this?

File: devicetypes/fibargroup/fibaro-double-switch-2-zw5.src/fibaro-double-switch-2-zw5.groovy [hunk 2]
```diff
@@ -329,27 +329,29 @@ def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cm
...
-				sendEvent([name: "energy", value: cmd.scaledMeterValue, unit: "kWh"])
+				sendEvent([name: "energy", value: value, unit: "kWh"])
 				break
 			case 2:
-				sendEvent([name: "power", value: cmd.scaledMeterValue, unit: "W"])
+				sendEvent([name: "power", value: value, unit: "W"])
 				break
 		}
-		multiStatusEvent("${(device.currentValue("power") ?: "0.0")} W | ${(device.currentValue("energy") ?: "0.00")} kWh")
+		multiStatusEvent("${(device.currentValue("power")?.doubleValue() ?: defaultValue)} W | ${(device.currentValue("energy")?.doubleValue() ?: defaultValue)} kWh")
```
Comment (tpmanley) [inline devicetypes/fibargroup/fibaro-double-switch-2-zw5.src/fibaro-double-switch-2-zw5.groovy:343]: This DTH is missing the definition of `attribute "multiStatus", "string"` so may as well fix that while you're in here.

**Review by tpmanley — APPROVED** [review]

**Comment by tpmanley** [comment]
Also, can you make the PR title more descriptive of the change, for example `[ICP-9107] Fibaro Dimmer Switch 2: Fix type of energy/power events`. That makes it much easier for us to see at a glance what the change is doing which is beneficial both for us internally and for anyone external who is looking at these changes.

**Comment by greens** [comment]
Cert team is now reporting that they could not reproduce the issue, so I don't know whether we should merge this.

**Pull request closed without merging**